go/consensus/tendermint/apps/roothash/transactions.go

```diff
@@ -140,6 +136,23 @@ func (app *rootHashApplication) executorCommit(
 		return err
 	}
 
+	if ctx.IsCheckOnly() {
+		for i, commit := range cc.Commits {
+			if !commit.NodeID.Equal(ctx.TxSigner()) {
+				// we wouldn't be able to handle this...
```

Comment:
I now realize we have the commit method that allows any transaction sender to send on behalf of any executor. so really that idea from before of "sender would have been registered anyway" doesn't even hold.

Reply:
Yes that is true, however currently the commits are not propagated otherwise so it will always be the node who submits them. We could however allow any node in the committee to submit commits (not just for itself).

Also note that this should then also be enforced during execution as now a bad proposer can bypass it.

Reply:
removing this from the pr, so gonna resolve this. let's keep it in mind if we later need exempt methods after all